android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java

```diff
@@ -2147,8 +2220,19 @@ protected void sendRPCMessagePrivate(RPCMessage message) throws SdlException {
 			pm.setMessageType(MessageType.RPC);
 			pm.setSessionType(SessionType.RPC);
 			pm.setFunctionID(FunctionID.getFunctionId(message.getFunctionName()));
-			pm.setPayloadProtected(message.isPayloadProtected());
-			
+			if (mRPCSecuredServiceStarted && mEncryptedRPCNames.contains(message.getFunctionName())) {

```

Comment:
Would it be possible to trigger `onError` for the RPC response listener if the RPC is a request and if the encryption service is not ready? maybe something like this:

```
if (mRPCSecuredServiceStarted && mEncryptedRPCNames.contains(message.getFunctionName())) {
	pm.setPayloadProtected(true);
} else {
	pm.setPayloadProtected(false);
}

if (pm.getPayloadProtected() && !mRPCSecuredServiceStarted){
	String errorMsg = "Trying to send an encrypted message and there is no secured service";
	if (message.getMessageType().equals(RPCMessage.KEY_REQUEST)) {
		RPCRequest request = (RPCRequest) message;
		request.getOnRPCResponseListener().onError(request.getCorrelationID(), Result.ABORTED, errorMsg);
	}
	Log.e(TAG, errorMsg);
}
```

Reply:
Done